pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -171,6 +171,21 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 
 //nolint:gocyclo
 func (c *Controller) finalize(ctx context.Context, nodeClaim *v1.NodeClaim) (reconcile.Result, error) {
+	// perform a no-op for whatever the status condition is currently set to
+	// so that we bump the observed generation to the latest and prevent the nodeclaim
+	// root status from entering an `Unknown` state
+	stored := nodeClaim.DeepCopy()
+	for _, condition := range nodeClaim.GetConditions() {
+		if condition.ObservedGeneration != nodeClaim.Generation {
+			nodeClaim.StatusConditions().Set(condition)
```

Comment:
Seems like you can just call set and not even do the check -- just blindly setting should be fine since if there is not change we just won't have to patch